### PR TITLE
Compliance contract

### PR DESCRIPTION
**Pull request by vistra**
With basic unit tests.
Does not notify other contracts on compliance change.

### CODE REVIEW

File: contracts/Compliance.sol [hunk 1]
```diff
@@ -0,0 +1,40 @@
+pragma solidity 0.5.16;
+
+import "@openzeppelin/contracts/ownership/Ownable.sol";
+import "./spec_interfaces/ICompliance.sol";
+
+contract Compliance is ICompliance, Ownable {
+
+    IContractRegistry contractRegistry;
```
Comment (ronnno) [inline contracts/Compliance.sol]: Why does it need a contract registry if nobody is using it?

and, if we later decide that this is a requirement for all contracts then maybe we should separate this into an interface or a super implemenration such as `RegisteredContract`: 
```
event ContractRegistryChanged(oldRegistry, newRegistry);

setContractRegistry(newRegisrtry)
getRegistryName() returns string
isRegistered() returns bool
...
```
and all our contracts can inherit from this contract..?

but if we don't have this standard, why not exclude this to avoid questions like - what kind of event is fired when we replace the regisrty? etc...

@OdedWx, @vistra what do you think ?

File: contracts/Compliance.sol [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    mapping (address => string) validatorCompliance;
+
+    /*
+     * External methods
+     */
+
+    function getValidatorCompliance(address addr) external view returns (string memory conformanceType) {
+        string memory compliance = validatorCompliance[addr];
+        if (bytes(compliance).length == 0) {
+            compliance = "General"; // TODO should this be configurable?
```
Comment (ronnno) [inline contracts/Compliance.sol]: The word compliance imply multiple string keys mapping to boolean values. such as: `compliance[addr][identified] == true`.

If we want to be specific, lets not call it compliance, it can be ValidatorIdentification contract. and the value can be boolean. If we are generic lets be generic because compliance is not a single string value you can comply to different things: GDPR compliant for instance, etc.. 

in other words, if you are "Identified" it does not mean you are not "General"...


@OdedWx, @vistra  ?

File: test/compliance.spec.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+describe('compliance-contract', async () => {
+
+    it('sets, gets, and updates validator compliance type', async () => {
+        // TODO see that committees are updates as a result of changing compliance
+
+        const d = await Driver.new();
+
+        const General = "General";
+        const Compliance = "Compliance";
```
Comment (ronnno) [inline test/compliance.spec.ts]: These constants can be put in the driver. they will be needed by other tests in the Elections contract

File: contracts/Compliance.sol (hunk 1 above)
Comment (vistra) [reply contracts/Compliance.sol]: It will be used when this contract sends notifications on compliance changes.
Agree we can use a super implementation, but not sure about the specifics. Anyway this is out of scope for this PR I think (this pattern is used across all contracts).

File: test/compliance.spec.ts (hunk 3 above)
Comment (vistra) [reply test/compliance.spec.ts]: RIght

File: contracts/Compliance.sol [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        validatorCompliance[addr] = conformanceType; // TODO should we only allow a predefined set? (i.e. "General", "Compliance")
+        emit ValidatorConformanceUpdate(addr, conformanceType);
+        // TODO notify elections contract?
+    }
+
+    /*
+     * Governance
+     */
+
+    function setContractRegistry(IContractRegistry _contractRegistry) external onlyOwner {
```
[the remark this answers is not in the document]
Comment (vistra) [reply contracts/Compliance.sol]: I think we will later move this logic to a super contract like you suggested and then decide if we need an event there.

[the remark this answers is not in the document]
Comment (ronnno) [reply contracts/Compliance.sol]: IMHO better not to leave such loose ends. if you want to postpone a comment is due, and I'd ask @OdedWx if he can quickly decide

File: contracts/Compliance.sol (hunk 1 above)
Comment (ronnno) [reply contracts/Compliance.sol]: why not send the notifications now?

**Comment by ronnno** [comment]
there are two questions here for @OdedWx and one question about merging with or without the notifications implemented. 
imho we better implement the notification so the contract is complete, unless I'm not understanding a technical difficulty?

File: contracts/Compliance.sol (hunk 4 above)
[the remark this answers is not in the document]
Comment (vistra) [reply contracts/Compliance.sol]: I'll add a todo

File: contracts/Compliance.sol (hunk 1 above)
Comment (vistra) [reply contracts/Compliance.sol]: I didn't want to touch the elections contract to avoid collisions with you as we discussed. I also prefer not adding nops as they create an illusion of integration. We will have it in the next 3 days anyway.

File: contracts/Compliance.sol (hunk 2 above)
Comment (vistra) [reply contracts/Compliance.sol]: Sounds good to me, I'll add a TODO for us to decide later. I prefer not to block the merge over this.

**Comment by vistra** [comment]
I prefer merging now with TODOs. Most of them will be gone anyway within the next few days.

File: contracts/Compliance.sol (hunk 4 above)
[the remark this answers is not in the document]
Comment (OdedWx) [reply contracts/Compliance.sol]: The interfaces miss all the governance events. Any change will rigger an event, similarly any change of managers will
